### PR TITLE
Solve for rho * u

### PR DESCRIPTION
**Pull request by jh66637**
The acoustic conservation equations read:
```math
 \frac{1}{c^2}\frac{\partial p}{\partial t} + \rho \nabla\cdot\mathbf{u} = f 
```
```math
\rho\frac{\partial \mathbf{u}}{\partial t}+\nabla p =\mathbf{0}
```
Similar to solving for the kinematic pressure in the incompressible Navier-Stokes equations we can solve for  $`\rho\mathbf{u}`$ instead of  $`\mathbf{u}`$:
```math
 \frac{1}{c^2}\frac{\partial p}{\partial t} + \nabla\cdot\mathbf{u}_{\rho} = f 
```
```math
\frac{\partial \mathbf{u}_{\rho}}{\partial t}+\nabla p =\mathbf{0}
```

This PR shifts the scaling factors to the temporal derivatives (as indicated in https://github.com/exadg/exadg/pull/619). Additionally, it restructures the operators such that we solve for  $`\rho\mathbf{u}`$. This PR should be merged before continuing with https://github.com/exadg/exadg/pull/619.

### CODE REVIEW

File: applications/acoustic_conservation_equations/vibrating_membrane/tests/skew_symmetric.output [hunk 1]
```diff
@@ -98,7 +98,7 @@ Calculate error for pressure at time t = 2.5300e-02:
   Absolute error (L2-norm): 1.56490e-02
 
 Calculate error for velocity at time t = 2.5300e-02:
-  Absolute error (L2-norm): 9.44139e-03
+  Absolute error (L2-norm): 2.83242e-02
```
Comment (jh66637) [inline applications/acoustic_conservation_equations/vibrating_membrane/tests/skew_symmetric.output:101]: The absolute error is exactly scaled by the density.

Comment (nfehn) [reply applications/acoustic_conservation_equations/vibrating_membrane/tests/skew_symmetric.output:101]: did you think about using relative errors?

File: include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp [hunk 2]
```diff
@@ -331,7 +331,9 @@ void
 SpatialOperator<dim, Number>::apply_inverse_mass_operator(BlockVectorType &       dst,
```
Comment (nfehn) [inline include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp:331]: This function name is now unclear in my opinion. `apply_inverse_mass_operator` should really be just the inverse mass. I could imagine we call this then `inverse_time_derivative_operator` (or `scaled_inverse_mass_operator`, with a documentation of what `scaled` means exactly in the context of two variables, i.e. velocity and pressure).

File: include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h [hunk 3]
```diff
@@ -200,9 +200,9 @@ class SoundEnergyCalculator
     CellIntegratorP pressure(matrix_free_in, dof_index_pressure, quad_index);
     CellIntegratorU velocity(matrix_free_in, dof_index_velocity, quad_index);
 
-    Number const rho       = static_cast<Number>(data.density);
+    Number const rho_inv   = static_cast<Number>(1.0 / data.density);
     Number const c         = static_cast<Number>(data.speed_of_sound);
-    Number const rhocc_inv = Number{1.0} / (rho * c * c);
+    Number const rhocc_inv = rho_inv / (c * c);
```
Comment (nfehn) [inline include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h:205]: I could not find `c` in the code below, so maybe this is better:

```C++
Number const rho_inv   = static_cast<Number>(1.0 / data.density);
Number const rhocc_inv = static_cast<Number>(1.0 / (data.density*data.speed_of_sound*data.speed_of_sound);
```

The variable name `rhocc_inv` would then also be clearer to me.

File: include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h [hunk 4]
```diff
@@ -222,7 +222,11 @@ class SoundEnergyCalculator
       {
         vector u = velocity.get_value(q);
```
Comment (nfehn) [inline include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h:223]: For clarity, variable name should be `rho_u` in my opinion

File: include/exadg/acoustic_conservation_equations/spatial_discretization/operators/operator.h [hunk 5]
```diff
@@ -371,28 +362,28 @@ class Operator
...
         pressure_m.submit_value(flux_momentum_weak, q);
-        velocity_m.submit_value(rho_inv * (flux_mass - pm) * n, q);
+        velocity_m.submit_value((flux_mass - pm) * n, q);
 
         if constexpr(weight_neighbor)
         {
           // minus signs since n⁺ = - n⁻
           pressure_p.submit_value(-flux_momentum_weak, q);
-          velocity_p.submit_value(-rho_inv * (flux_mass - pp) * n, q);
+          velocity_p.submit_value((pp - flux_mass) * n, q);
```
Comment (nfehn) [inline include/exadg/acoustic_conservation_equations/spatial_discretization/operators/operator.h:386]: Is there docu in the spatial operator that "velocity" describes `rho*u`? Maybe, writing `rho_u` explicitly here for the variable names would also be clearer here.

**Comment by nfehn** [comment]
I agree overall. I made several minor comments with the goal to make sure that it is easy to understand the code and which terms it implements.

File: include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp (hunk 2 above)
Comment (jh66637) [reply include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp]: [b7e5645](https://github.com/exadg/exadg/pull/631/commits/b7e564573a2e19ac1a69a3dc9ef2baab1755b1a4)

File: include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h (hunk 4 above)
Comment (jh66637) [reply include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h]: [caabe6f](https://github.com/exadg/exadg/pull/631/commits/caabe6f046a1795b4e88879d583d111aff8de496)

File: include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h (hunk 3 above)
Comment (jh66637) [reply include/exadg/acoustic_conservation_equations/postprocessor/sound_energy_calculator.h]: [caabe6f](https://github.com/exadg/exadg/pull/631/commits/caabe6f046a1795b4e88879d583d111aff8de496)

File: applications/acoustic_conservation_equations/vibrating_membrane/tests/skew_symmetric.output (hunk 1 above)
Comment (jh66637) [reply applications/acoustic_conservation_equations/vibrating_membrane/tests/skew_symmetric.output:101]: Yes, sadly in this particular test case, the analytical solution of the pressure and velocity are zero at certain times and therefore, we should use absolute errors here.

File: include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h [hunk 6]
```diff
@@ -163,9 +172,15 @@ class SpatialOperator : public Interface::SpatialOperator<Number>
                              BlockVectorType const & src,
                              double const            time) const;
 
-  // inverse mass operator
+  /**
+   * This function applys the inverse mass matrix and scales the pressure by c^2. This is because
```
Comment (nfehn) [inline include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h:176]: applies

File: include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h [hunk 7]
```diff
@@ -41,6 +41,15 @@ namespace ExaDG
...
 namespace Acoustics
 {
+/**
+ * Spatial operator to solve for the acoustic conservation equations:
+ *
+ * 1/c^2 * dp/dt + rho* div  u = f
+ *   rho * du/dt +      grad p = 0
+ *
+ * This operator solves for the pressure p as well as the velocity that is
+ * already scaled by the density rho * u.
```
Comment (nfehn) [inline include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h:51]: Given this description, I would in the equations above either skip rho, or write 
- `div (rho * u)` in the first equation, and 
- `d (rho * u)/dt` in the second equation.

**Review by nfehn — APPROVED** [review]
Great! Please slightly adjust the comments, then we are ready to merge.

File: include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h [hunk 8]
```diff
@@ -41,6 +41,15 @@ namespace ExaDG
 {
 namespace Acoustics
 {
+/**
+ * Spatial operator to solve for the acoustic conservation equations:
+ *
+ * 1/c^2 * dp/dt + div (rho * u) = f
+ *  (rho * u)/dt + grad p = 0
```
[the remark this answers is not in the document]
Comment (jh66637) [reply include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h]: https://github.com/exadg/exadg/commit/d9626a2e418930bbf56e548f17e1a867189be7f0

**Pull request merged**